R/return_linter.R

```diff
@@ -55,13 +71,19 @@
 #' @export
 return_linter <- function(
     return_style = c("implicit", "explicit"),
+    allow_implicit_else = TRUE,
     return_functions = NULL,
     except = NULL) {
   return_style <- match.arg(return_style)
 
+  if (!allow_implicit_else || return_style == "explicit") {
+    except_xpath <- glue("parent::expr[not(
+      preceding-sibling::expr/SYMBOL[{ xp_text_in_table(union(special_funs, except)) }]
```

Comment:
I was a bit worried about the `preceding-sibling::expr` and `parent::expr` here in the context of `=` or `->` assignment. Added a test for the former; the latter is not relevant I think, because of operator precedence:

```r
function() {
  ...
} -> .onLoad
```

Does not work as "intended" -- it actually parses like:

```r
function() ( foo <- { } )
```

So right-assignment of a function would actually have to look like:

```r
(function() {
  ...
}) -> .onLoad
```

That seems pretty unlikely to me, so ignoring it for now unless there's a user request.

Reply:
I do see a handful of such assignments, but I'm happy to ignore them, e.g.

https://github.com/cran/symbol.equation.gpt/blob/4c2d44d5f175f5ea5bbd540a1fd56d8cf0736f4f/R/aaa.R#L15-L22

https://github.com/cran/symbol.equation.gpt/blob/4c2d44d5f175f5ea5bbd540a1fd56d8cf0736f4f/R/main.R#L14-L20

https://github.com/search?q=lang%3AR+%2F%5B%7D%5D%5B%29%5D%5Cs*%5B-%5D%5B%3E%5D%2F+%2F%5E%5B%28%5D%5Cs*function%5Cs*%5B%28%5D%2F&type=code